eng/finalize-publish.yml

```diff
@@ -0,0 +1,58 @@
+parameters:
+  # Optional: dependencies of the job
+  dependsOn: ''
+
+  artifactsDir: $(Build.SourcesDirectory)/artifacts
+
+jobs:
+  - job: Finalize_Publish_Versions
+    displayName: Finalize_Publish_Versions
+    dependsOn: ${{ parameters.dependsOn }}
+    pool:
+      name: Hosted VS2017
+    steps:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download packages to publish
+      inputs:
+        artifactName: packages
+        downloadPath: ${{ parameters.artifactsDir }}
+
+    - script: InitTools.cmd
+      displayName: Restore Tools
+
+    - powershell: |
+        $prefix = "refs/heads/"
+        $branch = "$(Build.SourceBranch)"
+        $branchName = $branch
+        if ($branchName.StartsWith($prefix))
+        {
+          $branchName = $branchName.Substring($prefix.Length)
+        }
+        Write-Host "For Build.SourceBranch $branch, FullBranchName is $branchName"
+        Write-Host "##vso[task.setvariable variable=FullBranchName;]$branchName"
+        $(Build.SourcesDirectory)\\UpdatePublishedVersions.ps1 `
+        -gitHubUser dotnet-build-bot -gitHubEmail dotnet-build-bot@microsoft.com `
+        -gitHubAuthToken $(AccessToken-dotnet-build-bot-public-repo) `
+        -versionsRepoOwner dotnet -versionsRepo versions `
+        -versionsRepoPath build-info/dotnet/coreclr/$branchName `
+        -nupkgPath ${{ parameters.artifactsDir }}/packages/*.nupkg
```

Comment:
I don't think we place packages in `artifacts/packages` at the moment. They are in `bin/Product/<OS.Arch.Config>/.nuget/pkg`.

Reply:
Won't `Download packages to publish` put the packages here? Haven't used that yet, but it looks fine to me.

Reply:
Looks like build-job.yml is uploading like this
```
    # Upload build as pipeline artifact
    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
      - task: PublishPipelineArtifact@0
        displayName: Save product build as pipeline artifact
        inputs:
          artifactName: ${{ format('{0}_{1}_{2}_build', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
          targetPath: $(Build.SourcesDirectory)/bin/Product/$(osGroup).$(archType).$(buildConfigUpper)
    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
      - task: PublishPipelineArtifact@0
        displayName: Save product build as pipeline artifact
        inputs:
          artifactName: ${{ format('{0}_{1}_{2}_build', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
          targetPath: $(Build.SourcesDirectory)\bin\Product\$(osGroup).$(archType).$(buildConfigUpper)
```

I think the download is probably wrong....